src/visitor.rs

```diff
@@ -161,6 +154,13 @@ impl<'a> FmtVisitor<'a> {
             visit::FnKind::Closure => None,
         };
 
+        if let Some(ref single_line_fn) = self.rewrite_single_line_fn(&rewrite, &b) {
```

Comment:
Why do you have a separate `rewrite_single_line_fn` function and this check? I would have though we could just tweak `rewrite_fn` to write an empty block on one line - is that not possible?